### PR TITLE
[FIX] bi_sql_editor speed of manual field check

### PR DESCRIPTION
**Pull request by paulius-sladkevicius**
`check_manual_fields()` is hit quite a lot when loading addons. Improved by replacing a `startswith` with a faster slicing.

### CODE REVIEW

**Comment by paulius-sladkevicius** [comment]
@naglis could you take a look - before I submit it to OCA.

**Comment by paulius-sladkevicius** [comment]
BTW, this improved addon loading at least 2x.

File: bi_sql_editor/models/bi_sql_view.py [hunk 1]
```diff
@@ -730,9 +730,12 @@ def _refresh_size(self):
...
         # early on install / startup - particularly problematic during upgrade
-        if "group_operator" in table_columns(
-            self.env.cr, "bi_sql_view_field"
-        ) and model._name.startswith(self._model_prefix):
+        len_model_prefix = len(self._model_prefix)
+        if len(model._name) < len_model_prefix:
+            return
+        if model._name[:len_model_prefix] != self._model_prefix:
+            return
+        if "group_operator" in table_columns(self.env.cr, "bi_sql_view_field"):
```
Comment (naglis) [inline bi_sql_editor/models/bi_sql_view.py]: @paulius-sladkevicius I would suspect that the slow part would be the `table_columns()` call (that would explain the >2x query increase during module loading/testing), not the `startswith()` check. In such case, switching the operands should be enough, as then `and` would [short-circuit](https://docs.python.org/3/tutorial/datastructures.html?highlight=short%20circuit#more-on-conditions). IIUC that would be enough as the model prefix would not match in most cases.
```suggestion
        if model._name.startswith(self._model_prefix) and "group_operator" in table_columns(self.env.cr, "bi_sql_view_field"):
```

Comment (paulius-sladkevicius) [reply bi_sql_editor/models/bi_sql_view.py]: You're totally right, many thanks!

**Pull request closed without merging**